pkg/gateway/gateway.go

```diff
@@ -314,9 +344,17 @@ func (g *gatewayModule) SetFQDNProxy(wlID string, fqdn string, backends []string
 	if err := g.verifyDomainDestination(ctx, cfg, fqdn); err != nil {
 		return errors.Wrap(err, "failed to verify domain dns record")
 	}
-	return g.setupRouting(wlID, fqdn, backends, TLSPassthrough)
+	gatewayTLSConfig := TlsConfig{
+		CertResolver: httpCertResolver,
+		Domains: []Domain{
+			{
+				Main: fqdn,
+			},
+		},
+	}
+	return g.setupRouting(wlID, fqdn, backends, gatewayTLSConfig, TLSPassthrough)
 }
-func (g *gatewayModule) setupRouting(wlID string, fqdn string, backends []string, TLSPassthrough bool) error {
+func (g *gatewayModule) setupRouting(wlID string, fqdn string, backends []string, TLSConfig TlsConfig, TLSPassthrough bool) error {
```

Comment:
argument names always camel case with lower case initials

```suggestion
func (g *gatewayModule) setupRouting(wlID string, fqdn string, backends []string, tlsConfig TlsConfig, tlsPassthrough bool) error {
```